src/client/ui/viewcontrollers/EditorViewController.c

```diff
@@ -0,0 +1,329 @@
+			this->bumpSlider->delegate.didSetValue = didSetBump;
+
+			this->bumpSlider->min = 0.0;
+			this->bumpSlider->max = 20.0;
+			this->bumpSlider->step = 0.125;
+
+			$(this->bumpSlider, setValue, 1.0);
+
+			addInput((View *) stackView, "Bump", (Control *) this->bumpSlider);
+
+			// hardness
+
+			this->hardnessSlider = $(alloc(Slider), initWithFrame, NULL, ControlStyleDefault);
+
+			this->hardnessSlider->delegate.self = (ident *) this;
+			this->hardnessSlider->delegate.didSetValue = didSetHardness;
+
+			this->hardnessSlider->min = 0.0;
+			this->hardnessSlider->max = 20.0;
+			this->hardnessSlider->step = 0.1;
+
+			$(this->hardnessSlider, setValue, 1.0);
+
+			addInput((View *) stackView, "Hardness", (Control *) this->hardnessSlider);
+
+			// specular
+
+			this->specularSlider = $(alloc(Slider), initWithFrame, NULL, ControlStyleDefault);
+
+			this->specularSlider->delegate.self = (ident *) this;
+			this->specularSlider->delegate.didSetValue = didSetSpecular;
+
+			this->specularSlider->min = 0.0;
+			this->specularSlider->max = 20.0;
+			this->specularSlider->step = 0.1;
+
+			$(this->specularSlider, setValue, 1.0);
+
+			addInput((View *) stackView, "Specular", (Control *) this->specularSlider);
+
+			// parallax
+
+			this->parallaxSlider = $(alloc(Slider), initWithFrame, NULL, ControlStyleDefault);
+
+			this->parallaxSlider->delegate.self = (ident *) this;
+			this->parallaxSlider->delegate.didSetValue = didSetParallax;
+
+			this->parallaxSlider->min = 0.0;
+			this->parallaxSlider->max = 20.0;
+			this->parallaxSlider->step = 0.1;
+
+			$(this->parallaxSlider, setValue, 1.0);
+
+			addInput((View *) stackView, "Parallax", (Control *) this->parallaxSlider);
+
+			$((View *) box, addSubview, (View *) stackView);
+			release(stackView);
+
+			$((View *) column, addSubview, (View *) box);
+			release(box);
+		}
+
+		$((View *) columns, addSubview, (View *) column);
+		release(column);
+	}
+
+	$((View *) ((MenuViewController *) this)->panel->contentView, addSubview, (View *) columns);
+	release(columns);
+
+	((MenuViewController *) this)->panel->accessoryView->view.hidden = false;
+	addButton((View *) ((MenuViewController *) this)->panel->accessoryView, "Save", saveAction, self, NULL);
+
+	$(self->view, addSubview, (View *) ((MenuViewController *) this)->panel);
+
+	$((View *) this, updateBindings);
+}
+
+#pragma mark - EditorViewController
+
+/**
+ * @fn EditorViewController *EditorViewController::init(EditorViewController *self)
+ *
+ * @memberof EditorViewController
+ */
+static EditorViewController *init(EditorViewController *self) {
+
+	return (EditorViewController *) super(ViewController, self, init);
+}
+
+#pragma mark - Class lifecycle
+
+/**
+ * @see Class::initialize(Class *)
+ */
+static void initialize(Class *clazz) {
+
+	((ViewControllerInterface *) clazz->def->interface)->loadView = loadView;
+
+	((ViewInterface *) clazz->def->interface)->updateBindings = updateBindings;
```

Comment:
Yea, VC is not a View. This is going to explode :) No idea what method you're actually trodding on by casting to ViewInterface here, but it's not good.